mantidproject/mantid
# Fitting output now gives functions unique names.

**Description of work.**
This is a core change and would benefit from two reviews/tests. 

The output workspace from fitting currently has the; data, calculated, diff and then a break down of the functions. However, if multiples of the same function are used the name is not unique. This can prevent the users from generating the plots they want.
**Report to:** [user name]/[nobody]. 

**To test:**
Th following script adds a bunch of functions with tied parameters (we dont care about the actual fit just the output formatting). 

```
Load(Filename='HIFI00108371.nxs', OutputWorkspace='HIFI00108371') # can be any data
Fit(Function='name=Gaussian,Height=10,PeakCentre=15,Sigma=1,ties=(Height=10,PeakCentre=15,Sigma=1);name=FlatBackground,A0=1.2,ties=(A0=1.2);name=Gaussian,Height=12,PeakCentre=20,Sigma=1,ties=(Height=12,PeakCentre=20,Sigma=1);name=GausDecay,A=10,Sigma=0.2,ties=(A=10,Sigma=0.2);name=FlatBackground,A0=23,ties=(A0=23)', InputWorkspace='HIFI00108371', Output='HIFI00108371', OutputCompositeMembers=True, StartX=-3, EndX=33, Normalise=True)


```

when you look at the xxx_workspace you should see after diff a list of functions. Each function should be append with an appropriate number (for how many times that function has appeared). 



Fixes #24285. 




---
In release notes
#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
